File: Core/DriverHub.Application/Features/Entities/Reservations/Queries/GetReservationQuote/GetReservationQuoteResponse.cs
using DriverHub.Domain.Enums;

namespace DriverHub.Application.Features.Entities.Reservations.Queries.GetReservationQuote;

public sealed record GetReservationQuoteResponse(
    ReservationQuoteCarInfo Car,
    Guid PickupLocationId,
    string PickupLocationName,
    Guid ReturnLocationId,
    string ReturnLocationName,
    DateTime StartDate,
    DateTime EndDate,
    int RentalDays,
    decimal BasePrice,
    IReadOnlyCollection<ReservationQuoteExtraInfo> SelectedExtras,
    IReadOnlyCollection<ReservationQuoteExtraInfo> AvailableExtras,
    ReservationQuoteInsuranceInfo? SelectedInsurance,
    IReadOnlyCollection<ReservationQuoteInsuranceInfo> AvailableInsurancePackages,
    decimal ExtrasTotal,
    decimal InsurancePrice,
    decimal TotalPrice);

public sealed record ReservationQuoteCarInfo(Guid Id, string BrandName, string Model, short ModelYear, string Plate, string CoverImageUrl, CarStatus Status);
public sealed record ReservationQuoteExtraInfo(Guid Id, string Name, decimal DailyPrice);
public sealed record ReservationQuoteInsuranceInfo(Guid Id, string Name, decimal DailyPrice);

File: Client/DriverHub.Client/src/app/core/services/reservation/models/get-reservation-quote-response.ts
export interface GetReservationQuoteResponse {
  car: { id: string; brandName: string; model: string; modelYear: number; plate: string; coverImageUrl: string; status: number };
  pickupLocationId: string;
  pickupLocationName: string;
  returnLocationId: string;
  returnLocationName: string;
  startDate: string;
  endDate: string;
  rentalDays: number;
  basePrice: number;
  selectedExtras: ReservationExtraInfo[];
  availableExtras: ReservationExtraInfo[];
  selectedInsurance: InsuranceInfo | null;
  availableInsurancePackages: InsuranceInfo[];
  extrasTotal: number;
  insurancePrice: number;
  totalPrice: number;
}

export interface ReservationExtraInfo { id: string; name: string; dailyPrice: number; }
export interface InsuranceInfo { id: string; name: string; dailyPrice: number; }
